presto-main/src/main/java/com/facebook/presto/sql/relational/DeterminismEvaluator.java

```diff
@@ -57,9 +59,16 @@ public Boolean visitConstant(ConstantExpression literal, Void context)
         @Override
         public Boolean visitCall(CallExpression call, Void context)
         {
-            Signature signature = call.getSignature();
-            if (functionManager.isRegistered(signature) && !functionManager.getScalarFunctionImplementation(signature).isDeterministic()) {
```

Comment:
Why we wanna remove `isRegistered` interface?

Reply:
It's not really used much. The old implementation takes a signature, which is not sufficient when you have multiple FunctionNamespace. Instead of revamp the implementation, might as well remove it. After all, if resolve throws, it's not registered. ;)